packages/aws-cdk-lib/aws-sqs/lib/queue.ts

```diff
@@ -195,6 +202,32 @@ export interface DeadLetterQueue {
   readonly maxReceiveCount: number;
 }
 
+/**
+ * Permission settings for the dead letter source queue
+ */
+export interface SourceQueuePermission {
+  /**
+   * Permission settings for source queues that can designate this queue as their dead letter queue
+   *
+   * @default - `RedrivePermission.BY_QUEUE` if `sourceQueues` is specified; otherwise, `RedrivePermission.ALLOW_ALL`.
+   */
+  readonly redrivePermission?: RedrivePermission;
+
+  /**
+   * Source queues that can designate this queue as their Dead Letter Queue
+   *
+   * When RedrivePermission is set to BY_QUEUE, it is mandatory to configure sourceQueues.
+   *
+   * You can specify up to 10 source queues.
+   * To allow more than 10 source queues to specify dead-letter queues, set the `redrivePermission` to `ALLOW_ALL`.
+   *
+   * When either ALLOW_ALL or DENY_ALL is set in RedrivePermission, sourceQueues cannot be configured.
+   *
+   * @default - To not specify certain queues as source queues allowed for redrive.
```

Comment:
```suggestion
   * Source queues that can designate this queue as their dead-letter queue.
   *
   * When `redrivePermission` is set to `RedrivePermission.BY_QUEUE`, this parameter is required.
   *
   * You can specify up to 10 source queues.
   * To allow more than 10 source queues to specify dead-letter queues, set the `redrivePermission` to
   * `RedrivePermission.ALLOW_ALL`.
   *
   * When `redrivePermission` is either `RedrivePermission.ALLOW_ALL` or `RedrivePermission.DENY_ALL`,
   * this parameter cannot be set.
   *
   * @default - Required when `redrivePermission` is `RedrivePermission.BY_QUEUE`, cannot be defined otherwise.
```